### PR TITLE
Support Rails 8.0

### PR DESCRIPTION
**Pull request by a2ikm**
Note that Rails 8.0 requires Ruby >= 3.2.0.
https://rubygems.org/gems/rails/versions/8.0.0

### CODE REVIEW

File: lib/mongo_mapper/plugins/serialization.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def serializable_attributes
       end
 
       def serializable_hash(options = nil)
-        options ||= {}
+        options = options ? options.dup : {}
```
Comment (a2ikm) [inline lib/mongo_mapper/plugins/serialization.rb:21]: Fixed the following error:

```
  1) Serialization json should allow attribute only filtering
     Failure/Error: options[:only]   = Array.wrap(options[:only]).map { |k| k.to_s }
     
     FrozenError:
       can't modify frozen Hash: {:only=>[:age, :name]}
     # ./lib/mongo_mapper/plugins/serialization.rb:24:in `serializable_hash'
     # ./spec/unit/serialization_spec.rb:149:in `block (4 levels) in <module:Serialization>'
```

**Comment by a2ikm** [comment]
@smtlaissezfaire 
Hi, could you review, merge and release a new version that supports Rails 8.0? 😃

**Review by smtlaissezfaire — APPROVED** [review]
LGTM

**Comment by smtlaissezfaire** [comment]
0.17.0 released - https://mongomapper.com/blog/2024/12/10/release-0-17-0

**Comment by a2ikm** [comment]
Thank you!!

**Pull request merged**